v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreator.java

```diff
@@ -104,15 +135,15 @@ public void createShadowTablesInSpanner() {
    * Note: Shadow tables for interleaved tables are not interleaved to
    * their shadow parent table.
    */
-  Table constructShadowTable(Ddl informationSchema, String dataTableName, Dialect dialect) {
+  Table constructShadowTable(String dataTableName) {
 
     // Create a new shadow table with the given prefix.
     Table.Builder shadowTableBuilder = Table.builder(dialect);
     String shadowTableName = shadowTablePrefix + dataTableName;
     shadowTableBuilder.name(shadowTableName);
 
     // Add key columns from the data table to the shadow table builder.
-    Table dataTable = informationSchema.table(dataTableName);
+    Table dataTable = informationSchemaOfMainDb.table(dataTableName);
     Set<String> primaryKeyColNames =
         dataTable.primaryKeys().stream().map(k -> k.name()).collect(Collectors.toSet());
     List<Column> primaryKeyCols =
```

Comment:
I think the primary key builder logic is wrong, it is
`shadowTableBuilder.primaryKey().desc(keyColumn.name()).end()`, whereas it should first add all the columns and at last add `end()`

Reply:
The end is just for that primary key, this is being called in loop so this is okay. I have also tested this manually. Also the logic is similar in the forward migration template.